### PR TITLE
Alternative control scheme

### PR DESCRIPTION
**Pull request by hibyguy**
closes issue #65. In this implementation simply press the alternative control scheme button once for an overview of the new controls as well an actual control change. Then again to change them back to regular controls. For now its just a button that does some logic in the background to change the controls in a toggle like fashion; in the future(possibly this week) I'd like a pretty little checkbox. Currently there is no bugs with this pull just aesthetically I would like know what would fit best for this kind of change. Anyways this is my first bug fix, I figured I would pick a repo I actually like; so please, be as concise you want regarding edits to these changes. Thanks Ben! Thanks for letting me work on this repo 👍

### CODE REVIEW

**Comment by bfirsh** [comment]
Awesome! Thanks for the contribution.

I'm not sure I like the idea of a switch to a different set of controls. It seems very inelegant to me. Let me have a think though, because perhaps there is a better way.

I wonder how many people actually use the second player...

**Comment by hibyguy** [comment]
Thanks a bunch anyway! I needed to do a PR for my assignment! But to be honest I want to finish this to completion. Please tell me any changes you'd like from my code 💃 . Also I think anyone that uses a smaller laptop would appreciate this update :D (like me)

**Comment by hibyguy** [comment]
One of the ideas I had was adding a checkbox for alternative controls inside the display controls button

File: src/KeyboardController.jsx [hunk 1]
```diff
@@ -46,4 +72,8 @@ export default class KeyboardController {
   handleKeyPress = e => {
     e.preventDefault();
   };
+  
+  toggleControls = e => {
+      this.altButtons ? this.altButtons = false : this.altButtons = true;
```
Comment (bfirsh) [inline src/KeyboardController.jsx]:
```
this.altButtons = !this.altButtons;
```

File: src/KeyboardController.jsx [hunk 2]
```diff
@@ -20,23 +20,49 @@ const KEYS = {
...
       this.onButtonDown(key[0], key[1]);
       e.preventDefault();
     }
   };
 
   handleKeyUp = e => {
-    var key = KEYS[e.keyCode];
+	var key = null;
+	if(this.altButtons) key = ALT[e.keyCode];
+    else key = KEYS[e.keyCode];
```
Comment (bfirsh) [inline src/KeyboardController.jsx]: Inline if/else statements are error prone. I would recommend adding braces.

File: src/KeyboardController.jsx [hunk 3]
```diff
@@ -20,23 +20,49 @@ const KEYS = {
   100: [2, Controller.BUTTON_LEFT], // Num-4
   102: [2, Controller.BUTTON_RIGHT], // Num-6
 };
+const ALT = {
```
Comment (bfirsh) [inline src/KeyboardController.jsx]: `ALT_KEYS` is more descriptive, I think.

**Comment by hibyguy** [comment]
Amazing! I'll get started on this as soon as I can! I'm just swamped, thank you in advance for the opportunity! @bfirsh

File: src/KeyboardController.jsx (hunk 1 above)
Comment (bfirsh) [reply src/KeyboardController.jsx]: Come to think of it a better design would be `enableAltKeys` and `disableAltKeys`. Or, any kind of absolute state.

The risk with toggling is the UI toggle state could get out of sync with the internal toggle state, so the UI should always be the source of truth.

**Comment by bfirsh** [comment]
@hibyguy Thank you for the contribution! In the end I merged #136, which is more generic solution to the problem. Hope you don't take it the wrong way - I do appreciate your help. :)

If you're still up for helping out, there are a few ways that pull request could have been done better, which I noted in #149.

**Comment by hibyguy** [comment]
> @hibyguy Thank you for the contribution! In the end I merged #136, which is more generic solution to the problem. Hope you don't take it the wrong way - I do appreciate your help. :)
> 
> If you're still up for helping out, there are a few ways that pull request could have been done better, which I noted in #149.

Hey  its np thanks for giving me a chance! :)

**Pull request closed without merging**